### ISSUE TITLE
Editing names and tags of history items as soon as they are in the history

### ISSUE THREAD

**Issue by joachimwolff** [galaxyproject/galaxy#8417]
Hi,

To upload data can usually take a few hours to days and if ftp uploads via e.g. ENBI ENA are used, the name of the history item is just the URL. It would be a good feature to be able to edit the name and to add tags to an history item as soon as it appears in the history (grey status and yellow) and not just in the 'green' case. This would make it easier and reduce the overhead of renaming. If I just have selected to upload an item I usually know what it is and I can all do in one workflow. Currently, I have to come back after the upload (hours/days later) and have to figure out again which of the cryptic numbers are associated with which human-readable name.

Thanks for considering.

Best,

Joachim

**Comment by jdavcs** [comment]
I think tagging (but not name editing) should be already enabled during upload: https://github.com/galaxyproject/galaxy/blob/release_19.05/client/galaxy/scripts/mvc/dataset/dataset-li-edit.js#L123
The feature was added here: https://github.com/galaxyproject/galaxy/pull/7115
@jennaj : am I mistaken?

**Comment by jennaj** [comment]
Yes, tagging works. If the dataset flips state while typing those in, it get jumbled, but I don't see a way around that. 

BUT - the name cannot be changed. I would use that ALL the time if was made possible as well.

Clarification: tags can be added once the dataset is in the history, not in the Upload tools, which of course would be one step closer to "better". Keeping track of datasets is ... complicated. Tags help so much. 

Dataset names can be modified in the Upload tool. 

Other "Get-Data" tools -- being able to tag/rename the dataset(s) directly on the tool forms would be awesome. That would take some UX design to think through the use cases.

Renaming/adding tags during FTP upload would be another big push forward. 

In short: Once you start any process that loads data, you want to be able to label it while you still remember what it is, then move on to whatever else you are doing, which could be more data loading or not.
